Src/library/net/network/socket/network_socket.e

```diff
@@ -309,33 +362,63 @@ feature {NONE} -- Implementation
 		deferred
 		end
 
+	nanoseconds_to_seconds (ns: NATURAL_64): INTEGER
+		local
+			n64: NATURAL_64
+		do
+			n64 := ns // one_second_in_nanoseconds
+			if n64 > {INTEGER}.max_value.to_natural_64 then
+				Result := {INTEGER}.max_value
+			else
+				Result := n64.to_integer_32
+			end
+		end
+
 feature {NONE} -- Constants
 
-	default_timeout: INTEGER = 20
+	default_timeout: INTEGER
 			-- Default timeout duration in seconds
 
-feature {NONE} -- Externals
-
-	c_set_sock_recv_timeout (a_fd, level: INTEGER; a_timeout_seconds: INTEGER)
-			-- C routine to set socket option `SO_RCVTIMEO' with `a_timeout_seconds' seconds.
-		external
-			"C"
+	one_second_in_nanoseconds: NATURAL_64 = 1_000_000_000
+
+	max_timeout_ns_value: NATURAL_64 = 2_147_483_647_999_999_999
+			-- Maximum value for a timeout in nanoseconds.
+			-- See `is_valid_timeout_ns` for explanation.
+
+feature -- Validation			
+
+	is_valid_timeout_ns (ns: NATURAL_64): BOOLEAN
+			--| Note: internally (C API), the ns timeout is splitted into:
+			--|  - a long value `tv_sec`: number of seconds
+			--|  - a long value `tv_usec`: number of microseconds (with struct timeval)
+			--|  or  long value `tv_nsec`: number of nanoseconds  (with struct timespec)
```

Comment:
Where do we use `timespec` in the C API, I only see `timeval`? If we do not use `timespec` I would remove the comment.

Reply:
For now, we don't, but in the future we should use timespec, that is the reason we adopted nanosecond as timeout unit.
So I would keep this in comment.